internal/controller/apiextensions/claim/connection.go

```diff
@@ -99,3 +99,17 @@ func (a *APIConnectionPropagator) PropagateConnection(ctx context.Context, to xr
 
 	return true, nil
 }
+
+// LocalConnectionSecretFor creates a connection secret in the namespace of the
+// supplied LocalConnectionSecretOwner, assumed to be of the supplied kind.
```

Comment:
We only use this once in the whole codebase so I just duplicated it here for now. We could go back to using the one in runtime once runtime's `LocalConnectionSecretOwner` and `ConnectionSecretOwner` doesn't require the type to have ESS support.

The `ConnectionSecretFor` helpers were never aware of ESS in the first place, but they expect a type that satisfies `ConnectionSecretOwner` as an argument, and we added the ESS getters and setters to that interface.